bsmetadata/preprocessing_utils.py

```diff
@@ -15,6 +15,31 @@
 """
 from abc import ABC, abstractmethod
 from typing import Dict, List, Optional
+from urllib.parse import unquote, urlsplit
+
+from bsmetadata.vendor.dateutil.src.dateutil.parser import ParserError, parse
+
+
+def get_path_from_url(url):
+    """get the `path` part of `url`, with %xx escapes replaced by their single-character equivalent"""
+    parts = urlsplit(url)
+    return unquote(parts.path)
+
+
+def parse_date(path):
+    try:
+        return parse(path, fuzzy=True, date_only=True)
+    except ParserError:
+        return None
+    except OverflowError:
+        # this happens sometimes, I don't know why, just ignore it
+        return None
+
+
+def remove_improbable_date(x):
```

Comment:
It's great to have that kind of sanity check! Any particular reason for choosing 1983 as the earliest possible date?